lib/recipes-data/src/lib/extract_recipedata_from_element.ts

```diff
@@ -0,0 +1,30 @@
+import type {Block} from "@guardian/content-api-models/v1/block";
+import {ElementType} from "@guardian/content-api-models/v1/elementType";
+
+
+export interface RecipeOutput {
+  recipeUId: string;
+  jsonBlob: string;
+  checksum?: string;
+}
+
+export function extractRecipeData(webUrl: string, block: Block): RecipeOutput[] {
+  const allRecipes = block.elements.filter(elem => elem.type === ElementType.RECIPE)
+  if (allRecipes.length != 0) {
+    return allRecipes.map(recp => parseJsonBlob(webUrl, recp.recipeTypeData?.recipeJson))
+  } else {
+    throw new Error("Error! No recipe is present in elements!")
+  }
+}
+
+function parseJsonBlob(canonicalId: string, recipeJson: string | undefined): RecipeOutput {
+  const recipeData = JSON.parse(recipeJson as string) as Record<string, unknown>
+  if (!recipeData.id) {
+    throw new Error("Error! No id present in recipeJson!")
+  } else {
+    return <RecipeOutput>{
+      recipeUId: `${canonicalId}-${recipeData.id as string}`,
+      jsonBlob: JSON.stringify(recipeData)
```

Comment:
personally I'd just return `recipeData as string` here - it's a bit cheaper as we don't have to put it all back together again from the parsed content

Reply:
`recipeData as string` is not working, it will be nicer to return whole `recipeJson` to `jsonBlob` (discussed with @jonathonherbert) and then to test this by comparing recipeJson and jsonBlob in test file.